### ISSUE TITLE
Using custom OpenSSL version 3.x when compiling Python 3.x - fails

### ISSUE THREAD

**Issue by own3mall** [python/cpython#121992]
# Bug report

TL;DR: See https://github.com/python/cpython/issues/121992#issuecomment-2371230247 for a workaround.

### Bug description:

I've got a server running Ubuntu 20.04 that I needed to upgrade OpenSSL on.

I downloaded the source for OpenSSL 3.0.8 and configured it as so:

`./config --prefix=/usr/local/ssl --openssldir=/usr/local/ssl share`

Then, I did a make and make install.

I next reloaded the ldconfig by:

```
echo "/usr/local/ssl/lib64" > /etc/ld.so.conf.d/openssl.conf
ldconfig -v
```

Great, that worked.  The system is now using OpenSSL 3.0.8, but I need Python to use that version too, so I downloaded the source for Python-3.12.4 and configured it like so:

```
./configure --enable-optimizations --with-openssl=/usr/local/ssl --with-ensurepip=install --with-openssl-rpath=/usr/local/ssl/lib64/

```

Make results in:

```
[ERROR] _hashlib failed to import: /root/Downloads/Python-3.12.4/build/lib.linux-x86_64-3.12/_[hashlib.cpython-312-x86_64-linux-gnu.so](http://hashlib.cpython-312-x86_64-linux-gnu.so/): undefined symbol: EVP_MD_get_type
[ERROR] _ssl failed to import: /root/Downloads/Python-3.12.4/build/lib.linux-x86_64-3.12/_[ssl.cpython-312-x86_64-linux-gnu.so](http://ssl.cpython-312-x86_64-linux-gnu.so/): undefined symbol: SSL_get1_peer_certificate
```
To fix it, I had to edit the makefile:

`nano Makefile`

Change

`OPENSSL_LDFLAGS=-L/usr/local/ssl/lib`

To:

`OPENSSL_LDFLAGS=-L/usr/local/ssl/lib64`

# TLDR Summary:

Is there a way the ./config script can set `OPENSSL_LDFLAGS` properly for newer versions of OpenSSL which use `lib64` folder rather than `lib`?

### CPython versions tested on:

3.10, 3.12

### Operating systems tested on:

Linux

**Comment by honglei** [comment]
For CentOS7/Openssl3.3.1/Python3.12.4

export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/local/lib64
pkg-config --libs openssl

./configure --enable-shared --enable-optimizations --with-openssl=/usr/local/ --with-openssl-rpath=/usr/local/lib64 --prefix=/opt/python312 LDFLAGS="-Wl,-rpath=/opt/python312/lib -L/usr/local/lib64"

**Comment by zanghu** [comment]
> [@honglei](https://github.com/honglei) I've also tried to build OpenSSL 3 and Python 3.12 under CentOS 7. I've installed both into ~/.local The following Python build configure commands were successful in my case.
> 
> ./configure --enable-optimizations --prefix=$INSTALL_PATH \
>             --with-openssl-rpath=auto PKG_CONFIG_PATH=$SSL_INSTALL_PATH/lib64/pkgconfig
> # OR
> ./configure --enable-optimizations --prefix=$INSTALL_PATH --with-openssl=$SSL_INSTALL_PATH \
>             --with-openssl-rpath=$SSL_INSTALL_PATH/lib64 LDFLAGS=-L$SSL_INSTALL_PATH/lib64
> For versions of CPython older than 3.10 `--with-openssl-rpath` option is not available:
> 
> ./configure --enable-optimizations --prefix=$INSTALL_PATH \
>             LDFLAGS=-Wl,-rpath=$SSL_INSTALL_PATH/lib64 PKG_CONFIG_PATH=$SSL_INSTALL_PATH/lib64/pkgconfig

Thanks! It works for me.

**Comment by picnixz** [comment]
I'm going to close this one as we have a working workaround (I think it would be too hard to make it automatic as it depends on the Linux distribution)
